dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/InterfaceCompatibleRegistryProtocol.java

```diff
@@ -60,11 +65,18 @@ protected URL getRegistryUrl(URL url) {
 
     @Override
     public <T> ClusterInvoker<T> getServiceDiscoveryInvoker(Cluster cluster, Registry registry, Class<T> type, URL url) {
-        registry = registryFactory.getRegistry(super.getRegistryUrl(url));
+        try{
+            registry = registryFactory.getRegistry(super.getRegistryUrl(url));
+        }catch (IllegalStateException e){
+            logger.warn("do not support service discovery, automatically switch to interface-level service discovery.");
```

Comment:
I think it would be better if we notify users which registry is not compatible.

Reply:
ok, i will modify it.